File: BackEnd_Learn/Java/citylife/src/main/java/com/leeseojune53/citylife/Service/auth/AuthDetailsService.java
package com.leeseojune53.citylife.Service.auth;

import com.leeseojune53.citylife.Exceptions.UserNotFoundExceptions;
import com.leeseojune53.citylife.Repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class AuthDetailsService implements UserDetailsService {
    private final UserRepository userRepository;

    @Override
    public AuthDetails loadUserByUsername(String username) throws UsernameNotFoundException {
        return userRepository.findByUserId(username)
                .map(
                        user->{
                            return new AuthDetails(user.getUserId(), user.getUserPw());
                        })
                .orElseThrow(UserNotFoundExceptions::new);
    }
}

File: BackEnd_Learn/kotlin/auth/src/main/kotlin/io/github/leeseojune53/auth/security/auth/AuthDetailsService.kt
package io.github.leeseojune53.auth.security.auth

import io.github.leeseojune53.auth.entity.UserRepository
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.security.core.userdetails.UserDetailsService

class AuthDetailsService(
    private val userRepository: UserRepository
): UserDetailsService {
    override fun loadUserByUsername(username: String): UserDetails {
        return userRepository.findById(username)
                    .orElseThrow()
    }
}